pkg/gitnessclient/gitness_client.go

```diff
@@ -0,0 +1,416 @@
+		apiURL = api.String() + "?" + v.Encode()
+	} else {
+		apiURL = api.String()
+	}
+	req, err := http.NewRequest("GET", apiURL, nil)
+	if err != nil {
+		return "", fmt.Errorf("error while making reuest: %s", err.Error())
+	}
+
+	req.Header.Set("accept", "application/json")
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", g.Token))
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("error while making request: %s", err.Error())
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("failed to get commits status code %d: ", resp.StatusCode)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("error while reading response: %s", err.Error())
+	}
+
+	lastCommit, err := getLastCommit(body)
+	if err != nil {
+		return "", fmt.Errorf("error while fetching last commit from list: %s", err.Error())
+	}
+
+	return lastCommit.Sha, nil
+
+}
+
+func (g *GitnessClient) GetRepoRef(url string) string {
+	repoUrl := strings.TrimSuffix(url, ".git")
+	parts := strings.Split(repoUrl, "/")
+	var path string
+	if parts[3] == "git" {
+		path = fmt.Sprintf("%s/%s", parts[4], parts[5])
+	} else {
+		path = fmt.Sprintf("%s/%s", parts[3], parts[4])
+	}
+
+	return path
+}
+func getLastCommit(jsonData []byte) (Commit, error) {
+	var commitsResponse CommitsResponse
+	err := json.Unmarshal(jsonData, &commitsResponse)
+	if err != nil {
+		return Commit{}, fmt.Errorf("json Unmarshling failed: %s", err.Error())
+	}
+
+	sort.Slice(commitsResponse.Commits, func(i, j int) bool {
+		return commitsResponse.Commits[i].Committer.When.Before(commitsResponse.Commits[j].Committer.When)
+	})
+
+	if len(commitsResponse.Commits) == 0 {
+		return Commit{}, fmt.Errorf("no commits found")
+	}
+
+	return commitsResponse.Commits[len(commitsResponse.Commits)-1], nil
+}
+
+func (g *GitnessClient) GetPr(repoURL string, prNumber uint32) (*PullRequest, error) {
+	repoRef := g.GetRepoRef(repoURL)
+
+	apiUrl, err := g.BaseURL.Parse(fmt.Sprintf("/api/v1/repos/%s/pullreq/%d", url.PathEscape(repoRef), prNumber))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse url : %w", err)
+	}
+
+	req, err := http.NewRequest("GET", apiUrl.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("accept", "application/json")
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", g.Token))
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, err
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var pr PullRequest
+	err = json.Unmarshal(body, &pr)
+	if err != nil {
+		return nil, err
+	}
+	refPart := strings.Split(repoRef, "/")
+	pr.GitUrl = GetGitnessCloneUrl(g.BaseURL.Host, refPart[0], refPart[1])
+	return &pr, nil
+}
+
+func GetGitnessCloneUrl(source, owner, repo string) string {
+
```

Comment:
No need for the extra newlines wrapping this return statement.

Also, you should send the protocol as an argument to this function. The protocol (http/https) should be inferred either from the API URL of the repo URL.